tests/mt_api/mt_api_test.py

```diff
@@ -0,0 +1,1141 @@
+    :param provider_language_combination: The combination of MT provider and source/target language
+    :param settings: The fixture providing the django settings
+    :param httpserver: The fixture providing the dummy http server used for faking the DeepL API server
+    :param caplog: The :fixture:`caplog` fixture
+    """
+
+    provider, SOURCE_LANGUAGE_SLUG, TARGET_LANGUAGE_SLUG = provider_language_combination
+
+    # Test for english messages
+    settings.LANGUAGE_CODE = "en"
+
+    # Setup a mocked DeepL API server with dummy response
+    setup_fake_deepl_api_server(httpserver)
+
+    # Redirect call aimed at the DeepL API to the fake server
+    settings.DEEPL_API_URL = f"http://localhost:{httpserver.port}"
+
+    # Setup MT API supported languages
+    setup_deepl_supported_languages(["de"], ["en-gb", "en-us"])
+    setup_google_translate_supported_languages(["en"], ["ar"])
+
+    # Log the user in
+    client, role = login_role_user
+
+    # Translate the pages
+    selected_ids = [2, 3, 6] if provider == "DeepL" else [19]
+    machine_translation = reverse(
+        "machine_translation_pages",
+        kwargs={
+            "region_slug": REGION_SLUG,
+            "language_slug": TARGET_LANGUAGE_SLUG,
+        },
+    )
+
+    with patch.object(
+        GoogleTranslateApiClient, "__init__", setup_fake_google_translate_api
+    ):
+
+        response = client.post(
+            machine_translation, data={"selected_ids[]": selected_ids}
+        )
+        print(response.headers)
+
+        if role in PRIV_STAFF_ROLES + [AUTHOR, MANAGEMENT, EDITOR]:
+            # If the role should be allowed to access the view, we expect a successful result
+            assert response.status_code == 302
+            page_tree = reverse(
+                "pages",
+                kwargs={
+                    "region_slug": REGION_SLUG,
+                    "language_slug": TARGET_LANGUAGE_SLUG,
+                },
+            )
+            assert response.headers.get("Location") == page_tree
+            response = client.get(page_tree)
+
+            # Get the page objects including their translations from the database
+            page_translations = get_content_translations(
+                Page, selected_ids, SOURCE_LANGUAGE_SLUG, TARGET_LANGUAGE_SLUG
+            )
+            # Check that the success message are present
+            if TARGET_LANGUAGE_SLUG == "en":
+                assert_message_in_log(
+                    'SUCCESS  The following pages have successfully been translated (German ➜ English): "Über die App Integreat Augsburg", "Willkommen in Augsburg" and "Kontakt zu App Team Augsburg"',
+                    caplog,
+                )
+            else:
+                assert_message_in_log(
+                    'SUCCESS  Page "Learn German on Your Own" has successfully been translated (English ➜ Arabic).',
+                    caplog,
```

Comment:
I'd personally prefer it if the ids could be changed without changing the test. One way to do that would be to build the expected message dynamically, or if you decide to add the ids to the function parameters, you could also add the expected messages as parameters. But if you prefer it the way it is right now, I'm also fine with that :)

Reply:
Nice idea 👍 Applied (or at least I tried 🙈 )